cocotb/bfmgen.py

```diff
@@ -0,0 +1,364 @@
+###############################################################################
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+###############################################################################
+import os
+import argparse
+import importlib
+from string import Template
+
+import cocotb
+from cocotb.bfms import BfmMgr
+
+
+def bfm_load_modules(module_l): 
+    for m in module_l:
+        try:
+            importlib.import_module(m)
+        except Exception as e:
+            print("Error: failed to load module \"" + str(m) + "\": " + str(e))
+            raise e
```

Comment:
This doesn't seem like a good idea, the caller should log this exception anyway.

If you really want to keep it, can you change it to use a logger instead?

Reply:
Hmm... The caller is the bfmgen command-line script. I suppose we could just let the exception propagate back up, but I was concerned that the user wouldn't get appropriate information about which module had failed to load. Happy to change this if you feel the default Python exception trace will be better.

Reply:
Doesn't the exception information tell them which one failed anyway?

Reply:
It looks like this:
```
Traceback (most recent call last):
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/project/fun/cocotb/cocotb-task-bfms/cocotb/bfmgen.py", line 354, in <module>
    main()
  File "/project/fun/cocotb/cocotb-task-bfms/cocotb/bfmgen.py", line 348, in main
    bfm_load_modules(args.m)
  File "/project/fun/cocotb/cocotb-task-bfms/cocotb/bfmgen.py", line 27, in bfm_load_modules
    importlib.import_module(m)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'foobar'
```
So, yes, the missing module is there at the bottom. I'm concerned that it's unclear whether this is a user error (ie provoked by user input) or a system error (ie provoked by a bad installation). I'm find leaving it this way pending feedback if that's what seems to work best.